include: expose libuv version in header files

saghul opened the pull request:
Fix for #1111

/cc @indutny

Anyone feel free to provide feedback.

saghul committed:
include: expose libuv version in header files

sam-github commented:
Looks quite useful to me.

indutny commented:
LGTM, thank you!

bnoordhuis commented on include/uv-version.h:
```diff
@@ -0,0 +1,39 @@
+#define UV_VERSION_IS_RELEASE 0
+
+#endif /* UV_VERSION_H */
+
```
Newline at end.

bnoordhuis commented on include/uv-version.h:
```diff
@@ -0,0 +1,39 @@
+ /*
+ * Versions with an even minor version (e.g. 0.6.1 or 1.0.4) are API and ABI
+ * stable. When the minor version is odd, the API can change between patch
+ * releases. Make sure you update the -soname directives in config-unix.mk
```
This comment is slightly outdated. :-)

bnoordhuis commented:
Is there a reason to put the defines in a separate header file?

bnoordhuis commented on src/version.c:
```diff
@@ -19,21 +19,7 @@
  * IN THE SOFTWARE.
  */
 
-#include "uv.h"
```
You need to include uv.h here in order to pick up the right visibility for uv_version() and uv_version_string().  They won't be exported from the .so otherwise.

saghul replied:
Ah, right. Thanks Ben!
On Feb 15, 2014 11:50 AM, "Ben Noordhuis" notifications@github.com wrote:

> In src/version.c:
> 
> > @@ -19,21 +19,7 @@
> > - IN THE SOFTWARE.
> >   */
> > 
> > -#include "uv.h"
> 
> You need to include uv.h here in order to pick up the right visibility for
> uv_version() and uv_version_string(). They won't be exported from the .so
> otherwise.
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/joyent/libuv/pull/1114/files#r9772349
> .

saghul commented:
Addressed all feedback. About the separated file, @bnoordhuis: no specific reason, it felt right and I went for it, anything wrong with the approach?

Thanks!

saghul commented:
Thanks everyone, landed in https://github.com/saghul/libuv/commit/d4c12adff648bc4ebd5038e8186069193b22a693.

@piscisaureus, @tjfontaine you may need to update the release tool, since the version needs to be modified in a different file now.
